core/src/main/java/org/apache/calcite/sql/util/ChainedSqlOperatorTable.java

```diff
@@ -43,12 +42,12 @@
    * Creates a table based on a given list.
    */
   public ChainedSqlOperatorTable(List<SqlOperatorTable> tableList) {
-    this.tableList = ImmutableList.copyOf(tableList);
+    this.tableList = new ArrayList<>(tableList);
```

Comment:
Rather than relaxing the mutability of `tableList`, one option would be for callers to just construct a brand new ChainedSqlOperatorTable using the `of()` method if they wanted to add a new operator table.

Reply:
That was the initial approach but many tests failed as a result, I can look into it again though.